jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/RuleSetNodeInstance.java

```diff
@@ -72,32 +77,61 @@ public void internalTrigger(final NodeInstance from, String type) {
     	if ( !org.jbpm.workflow.core.Node.CONNECTION_DEFAULT_TYPE.equals( type ) ) {
             throw new IllegalArgumentException( "A RuleSetNode only accepts default incoming connections!" );
         }
-    	// first set rule flow group
-    	setRuleFlowGroup(resolveRuleFlowGroup(getRuleSetNode().getRuleFlowGroup()));
-    	
-    	//proceed
+    	RuleSetNode ruleSetNode = getRuleSetNode();
     	KnowledgeRuntime kruntime = getProcessInstance().getKnowledgeRuntime();
-    	Map<String, Object> inputs = evaluateParameters(getRuleSetNode());
-    	for (Entry<String, Object> entry : inputs.entrySet()) {
-    	    String inputKey = getRuleFlowGroup() + "_" +getProcessInstance().getId() +"_"+entry.getKey();
-    	    
-    	    factHandles.put(inputKey, kruntime.insert(entry.getValue()));
-    	}
+    	Map<String, Object> inputs = evaluateParameters(ruleSetNode);
     	
-    	if (actAsWaitState()) {
-        	addRuleSetListener();
-            ((InternalAgenda) getProcessInstance().getKnowledgeRuntime().getAgenda())
-            	.activateRuleFlowGroup( getRuleFlowGroup(), getProcessInstance().getId(), getUniqueId() );
-            
+    	if (ruleSetNode.isDMN()) {
+    	    String namespace = resolveVariable(ruleSetNode.getNamespace());
+    	    String model = resolveVariable(ruleSetNode.getModel());
+    	    String decision = resolveVariable(ruleSetNode.getDecision());
+    	    
+    	    DMNRuntime runtime = ((StatefulKnowledgeSession) kruntime).getKieRuntime(DMNRuntime.class);
+    	    DMNModel dmnModel = runtime.getModel(namespace, model);
+    	    if (dmnModel == null) {
+    	        throw new IllegalArgumentException( "DMN model '" + model + "' not found with namespace '" + namespace + "'");
+    	    }
```

Comment:
so if the builder has errors then KieContainerImpl will thrown an error when trying to getKieBase so I think we are good here, wdyt?

https://github.com/droolsjbpm/drools/blob/master/drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/KieContainerImpl.java#L568
https://github.com/droolsjbpm/drools/blob/master/drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/KieContainerImpl.java#L605